core/server/common/src/main/java/alluxio/master/journal/raft/JournalStateMachine.java

```diff
@@ -185,6 +187,17 @@ public StateMachineStorage getStateMachineStorage() {
         future.complete(reply);
         return future;
       }
+      if (queryRequest.hasAddQuorumServerRequest()) {
```

Comment:
This journal request can have various types of requests, like snapshot (above), and now add quorum? There should only be 1 set at a time?

Also, should this section be like:
```
if (queryRequest.hasGetSnapshotInfoRequest()) {
...
else if (queryRequest.hasGetSnapshotRequest()) {
...
else if (queryRequest.hasAddQuorumServerRequest()) {
...
}
```

Reply:
Yes there should be only one set for a request.

Both `GetSnapshotInfoRequest` and `GetSnapshotRequest` are handled by `SnapshotManager`. Only when a valid reply is not given does this function continue to check other types of request.

Reply:
I see. Can we just add a small comment, either here or above, saying something like handleRequest() handles GetSnapshotInfoRequest and GetSnapshotRequest, or returns `null`, so that this part handles the other requests (AddQuorumServerRequest).

Reply:
Added some comments.